### ISSUE TITLE
no-async-fn-without-await errors when await is last call

### ISSUE THREAD

**Issue by ekmartin** [avajs/eslint-plugin-ava#147]
```js
test.after(async () => {
  await app.stop();
});
```

Is this because the test could just return the promise explicitly instead? If so maybe it would be better to split it into its own rule, it's kind of confusing at the moment.

**Comment by sindresorhus** [comment]
That should not be an error. Sounds like a bug.

**Comment by jfmengels** [comment]
I agree, that sounds like a bug (though obviously, you could return the Promise directly).

I tried adding your example in my tests and was expecting tests to fail, but I could not reproduce.

Does that code excerpt (along with the require of `app` and `ava`) suffice to get that error report? If not, could you try to find a minimal viable bug to reproduce? Thanks!

**Comment by ekmartin** [comment]
It does indeed @jfmengels:

```js
import test from 'ava';

const something = () => null;
test.after(async () => {
  await something();
});
```

Using the newest release of xo (0.17), with `esnext` enabled.
